annorxiver_modules/document_helper.py

```diff
@@ -115,6 +136,102 @@ def generate_doc_vector(model, document_path, xpath, filter_tags=filter_tag_list
     return []
 
 
+def generate_doc_vector_parallel(
+    model, xpath, document_queue, vector_queue, filter_tags=filter_tag_list
+):
+    """
+    This method is designed to construct document vectors for a given xml document.
+    Every document has specific tags that are striped in order to have accurate embeddings
+
+    Keyword arguments:
+        model - the model to extract word vectors from
+        xpath_str - the xpath string to extract tags from the xml document
+        filter_tag_list - the list of tags to strip from the xml document
+    """
+    while True:
+
+        document_path = document_queue.get()
+
+        if document_path is None:
+            break
+
+        word_vectors = []
+
+        tree = ET.parse(open(document_path[2], "rb"), parser=parser)
+
+        # Process xml without specified tags
+        ET.strip_tags(tree, *filter_tags)
+
+        root = tree.getroot()
+        all_text = root.xpath(xpath)
+        all_text = list(map(lambda x: "".join(list(x.itertext())), all_text))
+        all_text = " ".join(all_text)
+
+        all_tokens = list(
+            map(
+                lambda x: x.lemma_,
+                filter(
+                    lambda tok: tok.lemma_ in model.wv
+                    and tok.lemma_ not in nlp.Defaults.stop_words,
+                    nlp(all_text),
+                ),
+            )
+        )
+
+        word_vectors += [model.wv[text] for text in all_tokens]
+
+        # skips weird documents that don't contain text
+        if len(word_vectors) > 0:
+            vector_queue.put(
+                (
+                    document_path[0],
+                    document_path[1],
+                    np.stack(word_vectors).mean(axis=0),
+                )
+            )
+
+    # Poison Pill to end writer
+    vector_queue.put(None)
+
+
+def write_document_vector_parallel(vector_queue, file_path_name, dim, n_jobs):
```

Comment:
Maybe add a description for this function

Reply:
Shoot I thought I wrote a description for this function. Good catch!